### ISSUE TITLE
why the loss has something*0

### ISSUE THREAD

**Issue by clearlovewl** [NVlabs/stylegan2-ada-pytorch#87]
why use (real_logits * 0 + loss_Dreal + loss_Dr1).mean().mul(gain).backward()  and (gen_img[:, 0, 0, 0] * 0 + loss_Gpl).mean().mul(gain).backward() .
it seems that ( loss_Dreal + loss_Dr1).mean().mul(gain).backward() and loss_Gpl.mean().mul(gain).backward() is simple, I tried and got different result, the first is better.
I guess it may be related with the pytorch gragh about autograd, could you tell me why?
